lumen/ai/agents.py

```diff
@@ -183,26 +183,22 @@ class SourceAgent(Agent):
             "Do NOT use if data sources already exist UNLESS user explicitly requests upload",
         ])
 
-    table_upload_callbacks = param.Dict(default={}, doc="""
-        Dictionary mapping from file extensions to callback function,
-        e.g. {"hdf5": ...}. The callback function should accept the file bytes and
-        table alias, add or modify the `source` in memory, and return a bool
-        (True if the table was successfully uploaded).""")
-
     purpose = param.String(default="The SourceAgent allows a user to upload new datasets, tables, or documents.")
 
     requires = param.List(default=[], readonly=True)
 
     provides = param.List(default=["sources", "source", "document_sources"], readonly=True)
 
+    source_controls: ClassVar[SourceControls] = SourceControls
```

Comment:
Thought it was really ugly to pass the source controls around as an instance var, so I made it a class var... I'd imagine most applications would want the same source controls?